Feature/vesting accounts support

atharen opened the pull request:
- Introduced spendable and locked balance calculations for vesting accounts.
- Updated UI to differentiate between total, spendable, and locked balances.
- Added logic to detect vesting accounts and load corresponding spendable balances.
- Improved overall balance display for both vesting and regular accounts.

Replaces #7

atharen committed:
Account balance display for vesting accounts

- Introduced spendable and locked balance calculations for vesting accounts.
- Updated UI to differentiate between total, spendable, and locked balances.
- Added logic to detect vesting accounts and load corresponding spendable balances.
- Improved overall balance display for both vesting and regular accounts.

atharen committed:
Enhance vesting account balance display

- Refined balance calculations for vesting accounts to include liquid and locked balances.
- Updated UI to reflect new balance categories: Liquid, Vesting, Delegation, Reward, and Unbonding.
- Added theoretical vested amount calculation based on vesting schedule.
- Improved clarity of balance descriptions and tooltips for better user understanding.

atharen committed:
feat(account): Refactor vesting account handling

- handling multi-vesting accounts with multiple schedule
- UI to display vesting schedule information, including progress.
- Refined balance display
- Added helper functions for better management of vesting account data.

atharen committed:
feat(account): Improve vesting account balances

- Balance calculations to leverages spendable and vesting amounts.
- Enhanced UI labels for clarity:
- 'Locked' to 'Vesting'
- 'Liquid' to 'Balance/Spendable'.
- Simplified balance rendering for maintainability.

atharen committed:
feat(chain): Update fuel mainnet configuration

- Reformatted JSON structure for better readability.
- Added asset details for FUEL including coingecko_id and logo.
- Specified features supported by the fuel mainnet.
- Removed redundant asset entry and ensured consistent formatting.

atharen committed:
readme: changelog for vesting account changes

changes related to support for vesting accounts and spendable balances

atharen committed:
fixup! feat(account): Improve vesting account balances

atharen committed:
fixup! fixup! feat(account): Improve vesting account balances

migueldingli1997 commented on src/modules/[chain]/account/[address].vue:
```diff
@@ -87,10 +109,305 @@ const totalValue = computed(() => {
   return format.formatNumber(value, '0,0.00');
 });
 
+// Calculate vesting percentage for vesting accounts
+const vestingPercentage = computed(() => {
```
We're calculating vested percentage here no? That's [how we use it](https://github.com/fuel-infrastructure/explorer/blob/cdc7a8adcdeab433beb6b611cf75ea0fbf4faa30/src/modules/%5Bchain%5D/account/%5Baddress%5D.vue#L786)

```suggestion
// Calculate vested percentage for vesting accounts
const vestedPercentage = computed(() => {
```

atharen replied:
refactored in 0e566326aff5ce90754414991478e4a9f97a3159

migueldingli1997 commented on src/modules/[chain]/account/[address].vue:
```diff
@@ -87,10 +109,305 @@ const totalValue = computed(() => {
+    }
+  }
+  
+  // Fallback for other vesting account types - use balance comparison
+  if (balances.value.length === 0 || spendableBalances.value.length === 0) {
+    return 0;
+  }
+  const totalAmount = Number(balances.value[0]?.amount || 0);
+  const spendableAmount = Number(spendableBalances.value[0]?.amount || 0);
+  
+  if (totalAmount === 0) return 0;
+  
+  return (spendableAmount / totalAmount) * 100;
```
Not sure what I think about this. It wouldn't be the correct percentage for vested if we use spendable. I'd rather return an `undefined`. Or just add a `console.warn` log.

atharen replied:
handled in 5349dcc

Preview should it be undefined.

![Screenshot 2025-06-13 at 14 37 47](https://github.com/user-attachments/assets/c34d1513-6091-4f8b-826e-c706ccbfad5a)

fyi, just forced the percentage to be undefined, but the rest work as usual.
didn't check if all info is missing.

migueldingli1997 commented on src/modules/[chain]/account/[address].vue:
```diff
@@ -87,10 +109,305 @@ const totalValue = computed(() => {
 });
 
+// Calculate vesting percentage for vesting accounts
+const vestingPercentage = computed(() => {
```
To simplify this function, what about using `theoreticalVestedAmount()` to get the vested amount instead of calculating it from scratch in this function. Then just divide it by the `totalOriginalVesting`.

migueldingli1997 replied:
You also have a `totalOriginalVesting` function actually. So you'd avoid calculating total original vesting in this function as well.

migueldingli1997 replied:
Be weary of dividing by zero

atharen replied:
handled in 35c54f2c8124548d71b84c910cf7e323ff824e62

preview if either `totalOriginalVesting` or `theoreticalVestedAmount` are undefined
![Screenshot 2025-06-13 at 16 32 02](https://github.com/user-attachments/assets/57f1b2d3-199a-4a6b-8ac7-c9321ebe0cae)

atharen replied:
The assets output is mildly improved in b09cdd1a64cd252ef33a2d3973c5a66f6091bda1
Can focus a bit more on it, but not worth the squeeze; like, data is missing anyway...

![Screenshot 2025-06-13 at 17 07 12](https://github.com/user-attachments/assets/5a0d2a13-c374-470c-a7d5-e9ed5afdbe60)

migueldingli1997 commented on src/modules/[chain]/account/[address].vue:
```diff
@@ -87,10 +109,305 @@ const totalValue = computed(() => {
+// Helper to get the primary denom from balances
+const primaryDenom = computed(() => {
+  return balances.value?.[0]?.denom || 'FUEL';
+});
```
The balances denom will be `fuel` and you're defaulting it to `FUEL`. There's a difference in unit size.

migueldingli1997 replied:
Also on testnet it would be `TEST`.

atharen replied:
just fallback on balance denom in 091e8669fba0fa89e5cb435051f07a730f8a3553

migueldingli1997 reviewed (changes requested):
Changes requested

atharen committed:
fixup! Account balance display for vesting accounts

atharen committed:
fixup! Account balance display for vesting accounts

atharen committed:
refactor(account): vesting calculations and UI

- helper function for vested amounts based on vesting schedules.
- theoretical vested amount and total original vesting handling undefined cases.
- Streamlined vesting calculations

atharen committed:
refactor(account): improve balance calculations with vesting

- particularly when vesting details are missing
- still not ideal, especially for totalValue, but mildly improved

atharen committed:
fixup! refactor(account): improve balance calculations with vesting

atharen committed:
refactor(account): enhance balance loading

- Introduced a separate function to load balances based on account type.
- Improved handling of spendable balances for vesting accounts, with fallback to regular balances on failure.
- Streamlined balance calculations in the totalValue computation.

atharen committed:
refactor(account): simplify primaryDenom computation
